lib/active_record_upsert/arel/crud.rb

```diff
@@ -12,9 +12,12 @@
 module Arel
   module Crud
     def compile_upsert(upsert_keys, upsert_options, upsert_values, insert_values, wheres)
+      # Support non-attribute key (like `md5(my_attribute)``)
+      target = upsert_keys.count == 1 && upsert_keys.first.class == ::Arel::Nodes::SqlLiteral ?
+          self[upsert_keys.first] : self[upsert_keys.join(',')]
```

Comment:
So, to follow up my own question - this is the place where "is it a hash?" would be the question, and it could use `::Arel::Nodes::SqlLiteral.new(upsert_keys[:literal])`